### PR TITLE
CIV-4338: When do you want to stop claiming interest?

### PR DESCRIPTION
**Pull request by kenneth-hmcts**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-4338

### Change description ###
This PR contains the changes for the page When do you want to stop claiming interest?

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by TatsyMaslova — APPROVED** [review]

**Review by RosDiaf — APPROVED** [review]

File: src/main/common/form/models/interest/interestEndDateSelection.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {IsDefined} from 'class-validator';
```
Comment (laarmada) [inline src/main/common/form/models/interest/interestEndDateSelection.ts:1]: Why that name and not interestEndDate.ts?

File: src/main/common/form/models/interest/interest.ts [hunk 2]
```diff
@@ -1,9 +1,15 @@
 import {InterestStartDate} from './interestStartDate';
+import {InterestEndDateType} from '../../../../common/form/models/claimDetails';
 
 export class Interest {
   interestStartDate?: InterestStartDate;
+  interestEndDate?: InterestEndDateType;
```
Comment (laarmada) [inline src/main/common/form/models/interest/interest.ts:6]: here should be interestEndDate.ts or current interestEndDateSelection.ts

File: src/main/routes/features/claim/interest/interestEndDateController.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+function renderView(form: GenericForm<InterestEndDateSelection>, res: Response): void {
+  res.render(interestEndDateViewPath, {form});
+}
+
+interestEndDateController.get(CLAIM_INTEREST_END_DATE_URL, async (req: AppRequest, res: Response, next: NextFunction) => {
+  try {
+    const claimId = req.session?.user?.id;
+    const interest = await getInterest(claimId);
+    const interestEndDate = Object.assign(new InterestEndDateSelection(), interest.interestEndDate);
```
Comment (laarmada) [inline src/main/routes/features/claim/interest/interestEndDateController.ts:24]: const interestEndDate = new InterestEndDateSelection(interest.interestEndDate));
we dont need to use assign

**Review by laarmada — CHANGES_REQUESTED** [review]
please improve the coverage: https://sonarcloud.io/summary/new_code?id=civil-citizen-ui&pullRequest=981

**Comment by commitBlob** [comment]
removing labels as no one is working on this PR at the moment.

**Review by laarmada — APPROVED** [review]

**Pull request merged**